name: Vale action

on: [pull_request]

permissions: {}

jobs:
  vale:
    name: runner / vale
    runs-on: ubuntu-latest
    permissions:
      contents: read
      pull-requests: write
    steps:
      - uses: actions/checkout@a5ac7e51b41094c92402da3b24376905380afc29 # v4.1.6
      - uses: errata-ai/vale-action@38bf078c328061f59879b347ca344a718a736018 # v2.1.0
        continue-on-error: true # GitHub Action flag needed until https://github.com/errata-ai/vale-action/issues/89 is fixed
        with:
          fail_on_error: true
          reporter: github-pr-check
          token: ${{secrets.GITHUB_TOKEN}}
